theano/gof/cmodule.py

```diff
@@ -208,7 +209,7 @@ def print_init(self, stream):
         print("}", file=stream)
 
     def add_include(self, str):
-        assert not self.finalized
+        # assert not self.finalized
```

Comment:
The problem is that the module_from_key() first calls the DynamicModule code function that set the finalized flag, after that it compiles the code and in that stage the add_include function is called and the assert not self.finalized fails.